dev/AppNotifications/ShellLocalization.cpp

```diff
@@ -167,10 +167,12 @@ void WriteHIconToPngFile(wil::unique_hicon const& hIcon, _In_ PCWSTR pszFileName
     THROW_IF_FAILED(spStreamOut->Commit(STGC_DANGEROUSLYCOMMITMERELYTODISKCACHE));
 }
 
-bool IsIconFileExtensionSupported(std::filesystem::path const& iconFilePath)
+bool Microsoft::Windows::AppNotifications::ShellLocalization::IsIconFileExtensionSupported(std::filesystem::path const& iconFilePath)
 {
     static PCWSTR c_supportedExtensions[]{ L".bmp", L".ico", L".jpg", L".png" };
 
+    winrt::check_bool(std::filesystem::exists(iconFilePath));
```

Comment:
There is a slight issue with doing this here.

With this approach, we return false if the file doesn't exist. However, the caller doesn't know if this is because of the icon being not found or if the file extension is unsupported.

Reply:
I would keep both operations in separate places (maybe do the exists check in the caller instead?).